componentLibraries/defaultLibrary/Signal/Sources&Sinks/SignalLogger.hpp

```diff
@@ -0,0 +1,260 @@
+            }
+            HVector<HString> namesVector = mNames.split(',');
+            HVector<HString> aliasesVector = mAliases.split(',');
+            HVector<HString> unitsVector =mUnits.split(',');
+
+            //Check that number of names/aliases/units are the same
+            //(only important for Hopsan row-wise CSV, where aliases and units are used)
+            if(mNumVars != namesVector.size()) {
+                stopSimulation("Number of variable names must equal number of input variables");
+                return;
+            }
+            if(mFileType == HopsanRowWiseCSV && mNumVars != aliasesVector.size()) {
+                stopSimulation("Number of variable aliases must equal number of input variables");
+                return;
+            }
+            if(mFileType == HopsanRowWiseCSV && mNumVars != unitsVector.size()) {
+                stopSimulation("Number of variable units must equal number of input variables");
+                return;
+            }
+
+            //Open file for writing
+            mFile.open(mFilePath.c_str());
+            if (!mFile.is_open()) {
+                stopSimulation("Could not open file for writing: "+ mFilePath);
+                return;
+            }
+
+            //Write header lines/rows depending on file type
+            if(mFileType == PlainColumnWiseCSV) {
+                //First row should be variable names
+                mFile << "Time,";
+                for(size_t i=0; i<mNumVars; ++i) {
+                    mFile << namesVector[i].c_str();
+                    if(i < mNumVars-1) {
+                        mFile << ",";
+                    }
+                }
+                mFile << "\n";
+            }
+            else if(mFileType == PlainRowWiseCSV) {
+                //First column should be variable names
+                //Create one row for each variable and insert names first
+                mRows.resize(mNumVars+1);
+                mRows[0].append("Time");
+                for(size_t i=0; i<mNumVars; ++i) {
+                    mRows[i+1].append(namesVector[i]);
+                }
+            }
+            else if(mFileType == HopsanRowWiseCSV) {
+                //First three columns should be variable names, aliases and units
+                //Create one row for each variable and insert this first
+                mRows.resize(mNumVars+1);
+                mRows[0].append("Time,,");
+                for(size_t i=0; i<mNumVars; ++i) {
+                    mRows[i+1].append(namesVector[i]+",");
+                    mRows[i+1].append(aliasesVector[i]+",");
+                    mRows[i+1].append(unitsVector[i]);
+                }
+            }
+            if(mFileType == HopsanPLO) {
+                //Write custom PLO header rows
+
+                //Extract target file name
+                HVector<HString> temp = mFilePath.split('/');
+                temp = temp[temp.size()-1].split('\\');
```

Comment:
Its unclear to me what happens here, does it split on both / and \, it looks like it extracts only the last character for the \\ split, maybe it already does, but it must support both / and \ file-paths

Reply:
No, temp is a `HVector`, so it extracts the last `HString`. The purpose is to first chop the string at the last forward slash, and then chop the remainder at the last backslash. In this way it extracts the file name regardless of the kind of slashes. It could maybe be problematic for filenames with spaces on Linux, such as "/home/user/my\ model.hmf", but I don't think this will be a major issue.

Reply:
Yes I see now what you mean